src/main/scala/spinoco/fs2/kafka/KafkaClient.scala

```diff
@@ -396,17 +396,22 @@ object KafkaClient {
       * @tparam F
       * @return
       */
-    def leaderFor[F[_] : Sync](
+    def leaderFor[F[_]: Sync : Log](
       requestMeta: (BrokerAddress, MetadataRequest) => F[MetadataResponse]
       , seed: Seq[BrokerAddress]
-    )(topicId: String @@ TopicName, partition: Int @@ PartitionId) :F[Option[BrokerAddress]] = {
+    )(topicId: String @@ TopicName, partition: Int @@ PartitionId): F[Option[BrokerAddress]] = {
       Stream.emits(seed)
       .evalMap { address => requestMeta(address, MetadataRequest(Vector(topicId))).attempt  }
-      .collect { case Right(response) => response }
-      .map { resp =>
-        resp.topics.find(_.name == topicId) flatMap { _.partitions.find( _.id == partition)} flatMap {
+      .flatMap {
+        case Left(err) => Stream.eval_(Log[F].error(s"Failed to read metadata for $topicId[$partition]", thrown = Some(err)))
```

Comment:
could we change the logging statement to Detail(topicId) and partition as we do in other places ?